modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/package.scala

```diff
@@ -57,8 +57,10 @@ package object engine {
     */
   def next(q: EventQueue): Engine[Unit] =
     gets(_.next).flatMap {
-      // No more Executions left
+      // Empty state
       case None     => send(q)(finished)
+      // Final State
+      case Some(qs@QStateF(_, _)) => put(qs) *> send(q)(finished)
```

Comment:
I was fiddling with the syntax but couldn't find a way to do what you just said! Thanks for the tip.